pandas/_libs/tslibs/strptime.pyx

```diff
@@ -288,20 +288,20 @@ def array_strptime(object[:] values, object fmt, bint exact=True, errors='raise'
             elif iso_year != -1 and iso_week != -1:
                 year, julian = _calc_julian_from_V(iso_year, iso_week,
                                                    weekday + 1)
-        # Cannot pre-calculate datetime_date() since can change in Julian
+        # Cannot pre-calculate datetime.date() since can change in Julian
         # calculation and thus could have different value for the day of the wk
         # calculation.
         try:
             if julian == -1:
                 # Need to add 1 to result since first day of the year is 1, not
                 # 0.
-                ordinal = datetime_date(year, month, day).toordinal()
-                julian = ordinal - datetime_date(year, 1, 1).toordinal() + 1
+                ordinal = datetime.date(year, month, day).toordinal()
```

Comment:
this is going to involve an extra python-space lookup, need to check perf

Reply:
This is the test case I have used for both of the tests:

```python
import numpy as np

import pandas._libs.tslibs.strptime as strptime

f = "%Y-%m-%d %H:%M:%S %Z"

arr = ['2010-01-01 12:00:00 UTC +0100'] * 100

strptime.array_strptime(np.array(arr, dtype="object"), f, exact=False)
```

---

master:
```
In [5]: %timeit strptime.array_strptime(np.array(arr, dtype="object"), f, exact=False)
1.19 ms ± 15.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

PR:
```
In [5]: %timeit strptime.array_strptime(np.array(arr, dtype="object"), f, exact=False)
1.2 ms ± 13.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```